apps/lib/publisher_agent/config.go

```diff
@@ -86,11 +84,18 @@ func LoadConfig(configFilePath string, keysFilePath string) (*StorkPublisherAgen
 	}
 
 	var keysFile KeysFile
-	err = json.Unmarshal(keyFileData, &keysFile)
-	if err != nil {
-		return nil, fmt.Errorf("failed to unmarshal keys file: %w", err)
+
+	// only deserialize keyFileData if keysFilePath was successfully read
+	if keyFileReadErr == nil {
```

Comment:
let's reorder this file a bit so we do all the loading for the config file followed by all the loading for the keys file.

feels a little weird to check for existence of an error a few lines later if we can avoid it - would be nice to just try to load the file and immediately take action on it